metals/src/main/scala/scala/meta/internal/metals/codeactions/ExtractValueCodeAction.scala

```diff
@@ -33,14 +33,19 @@ class ExtractValueCodeAction(
     val path = params.getTextDocument().getUri().toAbsolutePath
     val range = params.getRange()
 
-    val termOpt = trees.findLastEnclosingAt[Term](
-      path,
-      range.getStart(),
-      t => existsRangeEnclosing(t, range)
-    )
-
-    val textEdits = for {
-      term <- termOpt
+    val allTrees: Seq[Term] = {
+      def loop(t: Tree): Seq[Term] = {
+        t.children.find(_.pos.encloses(range)) match {
+          case Some(tr: Term) if existsRangeEnclosing(tr, range) =>
+            loop(tr) :+ tr
+          case Some(tr) => loop(tr)
+          case None => Nil
+        }
+      }
+      trees.get(path).map(loop(_)).getOrElse(Nil)
+    }
+    val textEdits: Seq[(Seq[l.TextEdit], String)] = for {
+      term <- allTrees
       stats <- lastEnclosingStatsList(term)
```

Comment:
I'm also thinking if we should put this and name generation above the for comprehension, because the result will be the same for every extract option